clickhouse-core-it/src/test/scala/xenon/clickhouse/HashSuite.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package xenon.clickhouse
+
+import xenon.clickhouse.base.ClickHouseSingleMixIn
+import xenon.clickhouse.client.NodeClient
+import xenon.clickhouse.hash._
+
+class HashSuite extends ClickHouseSingleMixIn with Logging {
+
+  def testHash[T <: Array[Any]](
```

Comment:
I'm afraid T could only be `Array[Any]` since `Array` is a final class 

```
final class Array[T](_length: Int)
```

BTW, `Array[String]` is not subclass of `Array[Any]`

Reply:
That's right, Changed to `Array[Any]`